### ISSUE TITLE
Issue with unpacking args in lambda function

### ISSUE THREAD

**Issue by jrdnh** [astral-sh/ty#1633]
### Summary

The following code raises an unexpected type error in the first snippet and does not raise a type error in the second snippet (which is invalid code).

Ideally would have hoped the required type for the arg could be inferred (#181) based on the context.

```python
# pow(x, y, modulo)
# type error on "pow", valid code
foo = lambda pair=(2, 2): pow(*pair, 4)
foo()

# no type error, causes runtime error
foo(('a', 'b'))
```

### Version

ty 0.0.1-alpha.27 (26d7b6864 2025-11-18)

**Comment by carljm** [comment]
Thanks for the report!

Our behavior here matches existing type checkers (pyright, mypy, and pyrefly), but I agree we would ideally prefer to do better.

This is (as you observe) closely related to #181, but it's a more "advanced" case that can't be handled with some (type-context-based) approaches to #181, so I'll keep it open as a sub-issue rather than closing as duplicate.
